js/Gameboard.js

```diff
@@ -1,23 +1,52 @@
-class Gameboard {
+  class Gameboard {
+    constructor() {
+      this.possibleCategories = this.generateCategories();
+    }
+
+  
+    const generateRandomIndex = (length) => {    
+      return Math.floor(Math.random() * length)
+    }
 
-  constructor() {
-    this.possibleCategories = this.generateCategories();
-  }
+    const chooseQuestion = (pointValue, categoryId, categoryIndex) => {
+    
+      const possibleQuestions = data.clues.filter(clue => {
+        return clue.pointValue === pointValue && clue.categoryId === categoryId
+      })
 
-  generateQuestions(currentRound) {
-    const gameboard = {
-      category1: [],
-      category2: []
-    //will generate categories and will generate questions from those categories.
+      const randomQuestion = possibleQuestions[generateRandomIndex(possibleQuestions.length)]
 
+      randomQuestion.pointValue *= (Math.floor((categoryIndex) / 4) + 1)
+
+      return randomQuestion;
     }
-    //return gameboard;
+
+    const shuffle = (a) => {
+      for (let i = a.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i + 1));
+        [a[i], a[j]] = [a[j], a[i]];
+      }
+      return a;
```

Comment:
This code is really difficult to read, can you explain what's happening here line-by-line?